pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -139,16 +137,66 @@ const (
 	// So the meta data can be erased when pushing to envoy.
 	PilotMetaKey = "pilot_meta"
 
-	// TODO(yxue): separate h2c vs h2
-	H2Protocol = "h2"
-
 	// CanonicalHTTPSPort defines the standard port for HTTPS traffic. To avoid conflicts, http services
 	// are not allowed on this port.
 	CanonicalHTTPSPort = 443
+
+	// Alpn HTTP filter name which will override the ALPN for upstream TLS connection.
+	AlpnFilterName = "istio.alpn"
 )
 
+type FilterChainMatchOptions struct {
+	// Application protocols of the filter chain match
+	ApplicationProtocols []string
+	// Transport protocol of the filter chain match. "tls" or empty
```

Comment:
This is not clear to me

Reply:
https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/listener/listener.proto

Reply:
It's used to denote the transport protocol for that filter chain.